yarpl/include/yarpl/flowable/Flowable.h

```diff
@@ -20,6 +20,19 @@
 namespace yarpl {
 namespace flowable {
 
+template <typename T>
+class Flowable;
+
+namespace detail {
+
+template <typename T>
+struct is_t_flowable : std::false_type {};
+
+template <typename R>
```

Comment:
Do we need all these or do we make the code more complex, do we have hard requirement for these?

template <typename T> , should we use T instead of R?

Reply:
Well the R references a different variable than T; T is `is_t_flowable<T>`, R is the template parameter that `T` is parameterized over in the case that `T` is a `Reference<Flowable<R>>`